File: FDP/T5_A8_PinedoBerumenClaraRuby/Ejercicio01_Calificaciones.py
'''
Created on 5 dic. 2019

@author: Ruby
'''

print("Programa que lee calificaciones\n")

numCal = int(input("Cuantas calificaciones desea ingresar?"))


def eliminar_repetidos(cals): 
    i=0
    while (i<len(cals)):
    #for i in range(0,(len(cals)-1)):
        j=i+1
        while(j<len(cals)):
        #for j in range(i+1,(len(cals))):
            if (cals[i]==cals[j]):
                del cals[j]
                #print(cals)
            j+=1
        i+=1       
    return cals           
                       
 
def leer_calificaciones():
    
    calificaciones=[]
    for i in range(numCal):
        calificaciones.append(float(input(f"Ingresa calificaion {i+1}:")))
        
    calificaciones=eliminar_repetidos(calificaciones)
    print(calificaciones)
    
    
       
        
leer_calificaciones()

File: FDP/T4_A2_PinedoBerumenClaraRuby/src/Ejercicio01_Calificaciones.java
import java.util.Scanner;

public class Ejercicio01_Calificaciones {

	public static void main(String[] args) {
		
		Scanner entrada = new Scanner(System.in);
		
		System.out.println("==Programa que pide calificaciones==");
		System.out.println();
		
		
		System.out.println("Cuantas calificaciones desea ingresar?");
		byte numcal = entrada.nextByte();
		
		int aprobatorias = 0;
		int reprobatorias = 0;
		int sumacal = 0;
		int numaprobatorias = 0;
		int prom = 0;
		int nummayorprom = 0;
		int summayorprom = 0;
		
		int calificaciones[];
		calificaciones = new int [numcal];
		
		for(int i=0; i<calificaciones.length; i++ ) {
			System.out.println("Ingresa calificación "+(i+1)+":");
			calificaciones[i] = entrada.nextInt();
		
			sumacal = sumacal + calificaciones[i];
			
			if(calificaciones[i]>=70) {
				numaprobatorias++;
				aprobatorias = aprobatorias+calificaciones[i];
				
			}else reprobatorias++;
			
               
		}//for
		prom = sumacal/numcal;
		

		for(int i=0; i<calificaciones.length; i++ ) {
			
		if(calificaciones[i]>=prom) {
			nummayorprom++;
			summayorprom=summayorprom + calificaciones[i];
		}
		 
		}
		
		
		
		System.out.println("Promedio general: " + prom);
		System.out.println("Promedio de calificaciones aprobatorias: "+ aprobatorias/numaprobatorias);
		System.out.println("Promedio de calificaciones mayores al promedio general: "
				+ summayorprom/nummayorprom);
		
		
		
		
	
		

	}

}
